Add upgrade steps (#6026)
wazuh/wazuh-documentation: source/deployment-options/docker/upgrading-wazuh-docker.rst

Upgrading Wazuh Docker
======================

This section describes how to upgrade a Wazuh Docker deployment starting from version 4.3. To upgrade Wazuh deployments with versions older than 4.3, refer to the :doc:`/deployment-options/docker/data-migration` documentation.

Below, you can see the steps for upgrading to version |WAZUH_CURRENT_MINOR|. These steps consider a scenario of a single node deployment. It considers that both, the ``docker-compose.yml`` and the mounted volumes, use their default names and configurations.

#. Change to the ``wazuh-docker/single-node/`` directory in the repository:

   .. code-block:: console

      # cd wazuh-docker/single-node

#. Run the following command to stop the outdated environment:

   .. code-block:: console

      # docker-compose down

#. Set the latest version of Wazuh in one of two ways.

   -  If you have a customized ``docker-compose.yml`` file and want to preserve it: Edit it and change the version of the images to the latest version.

>    -  If you want to use the default ``docker-compose.yml`` file of the latest version: Checkout the Git branch for this version.

Reviewer: ```suggestion
   -  If you want to use the default ``docker-compose.yml`` file of the latest version: Check out the Git branch for this version.
```

> This section describes how to upgrade Wazuh Docker deployments starting from version 4.3. To upgrade Wazuh deployments below version 4.3 (production deployment), refer to the :doc:`/deployment-options/docker/data-migration` documentation.

Reviewer: ```suggestion
This section describes how to upgrade Wazuh Docker deployments starting from version 4.3. To upgrade Wazuh deployments with versions below 4.3 (production deployment), refer to the :doc:`/deployment-options/docker/data-migration` documentation.
```

> This scenario is based on a single node deployment and both the docker-compose.yml and the mounted volumes have the default names and configuration.

Reviewer: ```suggestion
This scenario is based on a single node deployment, and both the docker-compose.yml and the mounted volumes have default names and configurations.
```

#. From the root directory of the repository ``wazuh-docker``, go to the single node deployment folder:

   .. code-block::

      # cd single-node

#. Run the following command to stop the current v4.3.x environment:

>    .. code-block::

Reviewer: ```suggestion
   .. code-block:: console
```

         # git checkout v|WAZUH_CURRENT_DOCKER|

> #. Before starting the new version, we must copy all the files that have been persisted in the docker volumes and belong to the Wazuh manager installation. This process will overwrite the files belonging to the Wazuh manager installation, but it will not delete your own customization files.

Reviewer: ```suggestion
#. Before starting the new version, we must copy all the files that have persisted in the docker volumes and belong to the Wazuh manager installation. This process will overwrite the files belonging to the Wazuh manager installation, but it will not delete your own customization files.
```

> #. First of all, we must comment out all the lines of the ``Docker-compose.yml`` file where docker volumes are mounted in the ``wazuh.manger`` and ``wazuh.indexer`` containers:

Reviewer: ```suggestion
#. First, you must comment out all the lines of the ``Docker-compose.yml`` file where docker volumes are mounted in the ``wazuh.manger`` and ``wazuh.indexer`` containers:
```

      # docker cp single-node_wazuh.manager_1:/var/ossec/var/multigroups/. - | sudo tar xf /dev/stdin -C $DOCKER_VOLUME_PATH/single-node_wazuh_var_multigroups/_data/
      # docker cp single-node_wazuh.manager_1:/var/ossec/integrations/. - | sudo tar xf /dev/stdin -C $DOCKER_VOLUME_PATH/single-node_wazuh_integrations/_data/
      # docker cp single-node_wazuh.manager_1:/var/ossec/active-response/bin/. - | sudo tar xf /dev/stdin -C $DOCKER_VOLUME_PATH/single-node_wazuh_active_response/_data/
      # docker cp single-node_wazuh.manager_1:/var/ossec/agentless/. - | sudo tar xf /dev/stdin -C $DOCKER_VOLUME_PATH/single-node_wazuh_agentless/_data/
      # docker cp single-node_wazuh.manager_1:/var/ossec/wodles/. - | sudo tar xf /dev/stdin -C $DOCKER_VOLUME_PATH/single-node_wazuh_wodles/_data/
      # docker cp single-node_wazuh.manager_1:/etc/filebeat/. - | sudo tar xf /dev/stdin -C $DOCKER_VOLUME_PATH/single-node_filebeat_etc/_data/
      # docker cp single-node_wazuh.manager_1:/var/lib/filebeat/. - | sudo tar xf /dev/stdin -C $DOCKER_VOLUME_PATH/single-node_filebeat_var/_data/

> #. If the directory inside the container has no files generates an error, dismiss it:

Reviewer: ```suggestion
#. Note that if the directory inside the container has no files, it will generate an error, and you must dismiss it:
```

   .. code-block:: bash

      # sudo docker cp single-node_wazuh.manager_1:/var/ossec/var/multigroups/. - | sudo tar xf /dev/stdin -C /var/lib/docker/volumes/single-node_wazuh_var_multigroups/_data/
      Error: No such container:path: single-node_wazuh.manager_1:/var/ossec/var/multigroups/.
      tar: This does not look like a tar archive
      tar: Exiting with failure status due to previous errors

> #. After the file copy, we must uncomment the previously commented lines in ``Docker-compose.yml`` file:

Reviewer: ```suggestion
#. After copying the file, we must uncomment the previously commented lines in the ``Docker-compose.yml`` file:
```

   .. code-block:: yaml
      :lineno-start: 25

      - wazuh_api_configuration:/var/ossec/api/configuration
      - wazuh_etc:/var/ossec/etc
      - wazuh_logs:/var/ossec/logs
      - wazuh_queue:/var/ossec/queue
      - wazuh_var_multigroups:/var/ossec/var/multigroups
      - wazuh_integrations:/var/ossec/integrations
      - wazuh_active_response:/var/ossec/active-response/bin
      - wazuh_agentless:/var/ossec/agentless
      - wazuh_wodles:/var/ossec/wodles
      - filebeat_etc:/etc/filebeat
      - filebeat_var:/var/lib/filebeat

   .. code-block:: yaml
      :lineno-start: 57

      - wazuh-indexer-data:/var/lib/wazuh-indexer

#. Start the new version of Wazuh using ``docker-compose``:

>    .. code-block::

Reviewer: ```suggestion
   .. code-block:: console
```
